spring-cloud-gcp-core/src/main/java/org/springframework/cloud/gcp/core/GcpEnvironment.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2017-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.core;
+
+/**
+ * Enumeration of valid individual GCP environments.
+ *
+ * <p>Also includes meta-members for convenient evaluation of environment groups, such as {@code ANY_APP_ENGINE}.
+ *
+ * @author Elena Felder
+ *
+ * @since 1.1
+ */
+public enum GcpEnvironment {
+
+	/**
+	 * Matches Kubernetes instances.
+	 */
+	KUBERNETES_ENGINE,
+
+	/**
+	 * Matches App Engine Flexible instances.
+	 */
+	APP_ENGINE_FLEXIBLE,
+
+	/**
+	 * Matches App Engine Standard instances.
+	 */
+	APP_ENGINE_STANDARD,
+
+	/**
+	 * Matches instances of Compute Engine that are not also AppEngine Flexible.
+	 */
+	COMPUTE_ENGINE,
+
+	/**
+	 * Matches both, AppEngine Flexible and AppEngine Standard.
+	 */
+	ANY_APP_ENGINE {
+		@Override
+		public boolean matchesSimpleEnvironment(GcpEnvironment env) {
```

Comment:
This seems a bit confusing. What does "simple" mean? I think that instead of `ANY_APP_ENGINE` and `GKE_OR_GCE`, it might be better to just have methods on the enum like `isAppEngine()` and `isGkeOrGce()`.

Reply:
I agree that it's confusing. But what would the rule look like then? Right now, it's
```
@ConditionalOnGcpEnvironment(GcpEnvironment.ANY_APP_ENGINE)
```

Reply:
That's a good point, but enums just don't lend themselves well to overlapping sets.
Can the annotation instead allow an array? 
```
@ConditionalOnGcpEnvironment([GcpEnvironment.APP_ENGINE_STANDARD, GcpEnvironment.APP_ENGINE_STANDARD])
```